### ISSUE TITLE
TreeQSM not creating QSM models in results...

### ISSUE THREAD

**Issue by Robin-hartley** [InverseTampere/TreeQSM#21]
Hi there,

I am getting the following error when I try and run individual tree point clouds in TreeQSM:

Error using save
Cannot create 'QSM_tree_t1_m1.mat' because 'results' does not exist.

Error in treeqsm (line 469)
          save(['results/QSM_',str],'QSM')

The code runs, it produces all of the figures for number of branches, branch order, volumes etc, however, it does not produce the QSM models. This same script was working fine up until about 2 months ago and then I tried running it one day and this error occurred. The 'results' folder is there but for whatever reason it is not being recognised. I am not very competent with MATLAB so any assistance in fixing this issue would be much appreciated as I am looking to do a lot of analysis with TreeQSM in the coming months.

Thanks in advance,

RJLH

**Comment by zoeschindler** [comment]
Hello :) Not my repository, but maybe I can help anyway. So if you have a results folder, it should usually work. Is the folder placed directly in you current working directory? You can get the current working directory with the command `pwd`. Another guess could be that your name has the wrong data type. In MATLAB, there are characters and strings. Some parts of MATLAB and also TreeQSM require characters, some require strings. I got it through trial and error, because I also don't usually use MATLAB for anything except TreeQSM.

**Comment by Robin-hartley** [comment]
Yep, that did it! thanks Zoe - I was running the code from another directory, so I just added a folder called "results" to the working directory and it is now writing the results.

Thanks heaps!

**Comment by sinianzuishuai** [comment]
> Hello :) Not my repository, but maybe I can help anyway. So if you have a results folder, it should usually work. Is the folder placed directly in you current working directory? You can get the current working directory with the command `pwd`. Another guess could be that your name has the wrong data type. In MATLAB, there are characters and strings. Some parts of MATLAB and also TreeQSM require characters, some require strings. I got it through trial and error, because I also don't usually use MATLAB for anything except TreeQSM.

Hi, I'm having the same problem with the string 'str', the second case you mentioned. Remind me that the variable 'str' seems to change with the number of iterations, please pre-allocate memory for higher speed. Do you know how to fix this, I'm struggling with it.
![pic](https://github.com/user-attachments/assets/d3732350-0144-492f-947c-db1d8d1ed6d3)

**Comment by sinianzuishuai** [comment]
> 你好 :) 不是我的存储库，但也许我可以帮忙。所以如果你有一个结果文件夹，它通常应该可以工作。文件夹是否直接放在你当前的工作目录中？你可以使用命令获取当前工作目录`pwd`。另一个猜测可能是你的名字有错误的数据类型。在 MATLAB 中，有字符和字符串。MATLAB 和 TreeQSM 的某些部分需要字符，有些需要字符串。我通过反复试验找到了答案，因为除了 TreeQSM 之外，我通常也不会将 MATLAB 用于任何其他用途。

Oh my gosh, I read what you said again, put it in the current directory, put it in the current directory, put it in the current directory ！！！！ Thank you so much dear
